src/Module.Server/Modes/DTV/CrpgDTVTeamSelectComponent.cs

```diff
@@ -0,0 +1,330 @@
+﻿using TaleWorlds.MountAndBlade;
+using Crpg.Module.Api.Models.Users;
+
+#if CRPG_SERVER
+using System.Text;
+using Crpg.Module.Balancing;
+using Crpg.Module.Common.Network;
+using NetworkMessages.FromClient;
+using TaleWorlds.Core;
+using TaleWorlds.Library;
+using TaleWorlds.PlayerServices;
+#endif
+
+namespace Crpg.Module.Common;
+
+/// <summary>
+/// Disables team selection and randomly assign teams if the native balancer is enabled. Else use the cRPG balancer
+/// to balance teams after each round.
+/// </summary>
+internal class CrpgDTVTeamSelectComponent : MultiplayerTeamSelectComponent
+{
+#if CRPG_SERVER
+    private readonly MultiplayerWarmupComponent _warmupComponent;
+    private readonly MultiplayerRoundController? _roundController;
+    private readonly PeriodStatsHelper _periodStatsHelper;
+
+    /// <summary>
+    /// Players waiting to be assigned to a team when the cRPG balancer is enabled.
+    /// </summary>
+    private readonly HashSet<PlayerId> _playersWaitingForTeam;
+
+    public CrpgDTVTeamSelectComponent(MultiplayerWarmupComponent warmupComponent, MultiplayerRoundController? roundController)
+    {
+        _warmupComponent = warmupComponent;
+        _roundController = roundController;
+        _periodStatsHelper = new PeriodStatsHelper();
+        _playersWaitingForTeam = new HashSet<PlayerId>();
+    }
+#endif
+
+    public override void OnBehaviorInitialize()
+    {
+        base.OnBehaviorInitialize();
+
+#if CRPG_SERVER
+        _warmupComponent.OnWarmupEnded += OnWarmupEnded;
+        if (_roundController != null)
+        {
+            _roundController.OnPostRoundEnded += OnRoundEnded;
+        }
+    }
+
+    public override void OnRemoveBehavior()
+    {
+        base.OnRemoveBehavior();
+        if (_roundController != null)
+        {
+            _roundController.OnPostRoundEnded -= OnRoundEnded;
+        }
+
+        _warmupComponent.OnWarmupEnded -= OnWarmupEnded;
+    }
+
+    public void SetPlayerAgentsTeam()
+    {
+        Debug.Print("Setting player agents' team");
+
+        DTVGameMatch gameMatch = TeamsToGameMatch();
+        DTVGameMatch balancedGameMatch = MovePlayersToDefenderTeam(gameMatch);
+
+        Dictionary<int, Team> usersToMove = ResolveTeamMoves(current: gameMatch, target: balancedGameMatch);
+        var crpgNetworkPeers = GetCrpgNetworkPeers();
+        SendSwapNotification(usersToMove, crpgNetworkPeers);
+
+        foreach (var userToMove in usersToMove)
+        {
```

Comment:
Why are you moving players automatically? You can just let them join the team and prevent joining the bots team